interpreter/python/decode_amd64.go

```diff
@@ -1,30 +1,163 @@
-//go:build amd64
-
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
 package python // import "go.opentelemetry.io/ebpf-profiler/interpreter/python"
 
 import (
-	"unsafe"
+	"fmt"
 
 	"go.opentelemetry.io/ebpf-profiler/libpf"
-	_ "go.opentelemetry.io/ebpf-profiler/zydis" // links Zydis
+	"golang.org/x/arch/x86/x86asm"
 )
 
-// #cgo CFLAGS: -g -Wall
-// #include "decode_amd64.h"
-// #include "../../support/ebpf/types.h"
-import "C"
+const debugDecodeAMD64 = false
+
+func regIndex(reg x86asm.Reg) int {
+	switch reg {
+	case x86asm.RAX, x86asm.EAX:
+		return 1
+	case x86asm.RBX, x86asm.EBX:
+		return 2
+	case x86asm.RCX, x86asm.ECX:
+		return 3
+	case x86asm.RDX, x86asm.EDX:
+		return 4
+	case x86asm.RDI, x86asm.EDI:
+		return 5
+	case x86asm.RSI, x86asm.ESI:
+		return 6
+	case x86asm.RBP, x86asm.EBP:
+		return 7
+	case x86asm.RSP, x86asm.ESP:
+		return 8
+	case x86asm.RIP:
+		return 9
+	default:
+		return 0
+	}
+}
+
+type regState struct {
+	loadedFrom uint64
+	value      uint64
+}
+
+func decodeStubArgumentAMD64(code []byte, codeAddress, memoryBase uint64) uint64 {
+	targetRegister := x86asm.RDI
+
+	instructionOffset := 0
+	regs := [32]regState{}
+
+	for instructionOffset < len(code) {
+		rem := code[instructionOffset:]
+		if len(rem) >= 4 &&
+			code[instructionOffset] == 0xf3 &&
+			code[instructionOffset+1] == 0x0f &&
+			code[instructionOffset+2] == 0x1e &&
+			code[instructionOffset+3] == 0xfa {
```

Comment:
So this is a instruction the disassembler does not support, but we need to recognize and skip? This should go to a helper module. But if preferred, I can do this when working with the other conversions and try to see what would be a nice api.